scenario.py

```diff
@@ -468,12 +471,19 @@ def plot_data_distribution(self):
         plt.xlabel("Digits")
         # plt.show()  # DEBUG
         plt.savefig(self.save_folder / "data_distribution.png")
+        plt.close()
 
     def compute_batch_sizes(self):
 
         # For each partner we compute the batch size in multi-partner and single-partner setups
+        if self.partners_count == 1:
+            p = self.partners_list[0]
+            p.batch_size = max(1, int(len(p.x_train) / self.gradient_updates_per_pass_count))
```

Comment:
Why don't we use the same batch size calculation when there is only one partner ?

Reply:
If there is only one partner in the scenario there is not going to be any multi-partner learning at all, only single partner learning cases. As a consequence there is no minibatch iterations, so it would not make sense to compute the `batch_size` based on the number of gradient updates per pass for the multi-partner learning cases. In this (rare) case, we compute the `batch_size` for enforcing the `gradient_updates_per_pass_count` in single partner learning case.

Reply:
My bad, I thought it was when doing contributivity computations on a subset of partners.
This is indeed a rare case and this not distributed learning but "standard" machine learning.